### PR TITLE
Rewrite documentation to be easier to get started.

### PR DESCRIPTION
**Pull request by fdobrovolny**
It also does general styling edits and typo fixes.

This commit will break documentation building as the pages for the menu change this will be solved by a PR to the docs repository which has to be merged simultaneously.

### CODE REVIEW

**Review by matofeder — APPROVED** [review]
LGTM! 
Just a few minor suggestions which may or may not be accepted.

File: doc/make-reference.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+### make attach
+
+``make attach RESOURCE=<resource-id> PARAMS=...``
+
+This will attach a resource to the terraform state. This is useful if you have created a resource outside of terraform
+and want to manage it with terraform.
+
+### make detach
+
+``make detachRESOURCE=<resource-id> PARAMS=...``
```
Comment (chess-knight) [inline doc/make-reference.md:156]:
```suggestion
``make detach RESOURCE=<resource-id> PARAMS=...``
```

**Review by chess-knight — APPROVED** [review]
Lgtm, just small typos/suggestions

File: doc/make-reference.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+Finally, a test cluster is created utilizing Kubernetes CAPI. This test cluster allows for experimentation, validation,
+and development within the CAPI environment.
+
+> Note that ``make create`` will not create a testcluster if you have set ``controller_count`` to zero in your
+> environment file (``environment-<yourcloud>.tfvars``).
+
+> Note that ``make create`` does not copy local files to the management server. If you want to change any of the scripts
+> being copied to the management server, you need to commit and push your changes and run ``make create`` again.
+> ``make create`` will pull the latest changes from the git repository.
```
Comment (fdobrovolny) [inline doc/make-reference.md:31]:
```suggestion
> Note that ``make create`` does not copy local files to the management server, only some files are templated there from the `terraform/files/template` directory. If you want to change any of the scripts
> being copied to the management server, you need to commit, push your changes, and run ``make create`` again.
> ``make create`` will pull the latest changes from the git repository.
```

File: doc/make-reference.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+``make get-kubeconfig``
+
+This will get the kubeconfig of the testcluster and store it in the file ``testcluster.yaml.<yourcloud>``.
+
+### make ssh
+
+``make ssh``
+
+This will ssh into the management server. You may need to set the ``USERNAME`` variable to the username you set
+in your ``environment-<yourcloud>.tfvars`` file. The default is ``ubuntu``.
```
Comment (fdobrovolny) [inline doc/make-reference.md:44]:
```suggestion
in your ``environment-<yourcloud>.tfvars`` file. The default in the environment file is ``ubuntu``.

> Note: there is also an alias to this `make login`
```

File: doc/make-reference.md [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+*all resources in the OpenStack project* even those that have not been created through this
+Terraform script or the Cluster API.
+It requires the [``ospurge``](https://opendev.org/x/ospurge) tool.
+Install it with ``python3 -m pip install git+https://git.openstack.org/openstack/ospurge``.
+
+## Testing commands
+
+### make check
+
+``make check``
```
Comment (fdobrovolny) [inline doc/make-reference.md:105]:
```suggestion
``make check SONOMODE=...``
```

File: doc/make-reference.md [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+
+This will run tests of the configuration on testcluster using [sonobuoy](https://sonobuoy.io/). It will also download the results and
+print them to the console.
```
Comment (fdobrovolny) [inline doc/make-reference.md:108]:
```suggestion
print them to the console. Optionally you can also specify a mode by using for example `SONOMODE="--mode quick"`
```

**Review by fdobrovolny — COMMENTED** [review]
@chess-knight Could you check these changes, please?

File: doc/make-reference.md (hunk 2 above)
Comment (chess-knight) [reply doc/make-reference.md:31]: Lgtm

File: doc/make-reference.md (hunk 3 above)
Comment (chess-knight) [reply doc/make-reference.md:44]: Lgtm

File: doc/make-reference.md (hunk 4 above)
Comment (chess-knight) [reply doc/make-reference.md:105]: Lgtm

**Pull request merged**